lib/fetch/body.js

```diff
@@ -397,7 +398,47 @@ function bodyMixinMethods (instance) {
 
       // If mimeType’s essence is "multipart/form-data", then:
       if (/multipart\/form-data/.test(contentType)) {
-        throw new NotSupportedError('multipart/form-data not supported')
+        const headers = {}
+        for (const [key, value] of this.headers) headers[key.toLowerCase()] = value
+
+        const responseFormData = new FormData()
+
+        let busboy
+
+        try {
+          busboy = Busboy({ headers })
+        } catch (err) {
+          // Error due to headers:
+          throw Object.assign(new TypeError(), { cause: err })
+        }
+
+        busboy.on('field', (name, value) => {
+          responseFormData.append(name, value)
+        })
+        busboy.on('file', (name, value, info) => {
+          const { filename, encoding, mimeType } = info
+          const base64 = encoding.toLowerCase() === 'base64'
+          const chunks = []
+          value.on('data', (chunk) => {
+            if (base64) chunk = Buffer.from(chunk.toString(), 'base64')
+            chunks.push(chunk)
+          })
+          value.on('end', () => {
+            const file = new File(chunks, filename, { type: mimeType })
+            responseFormData.append(name, file)
+          })
+        })
+
+        const busboyResolve = new Promise((resolve, reject) => {
+          busboy.on('finish', resolve)
+          busboy.on('error', (err) => reject(err))
```

Comment:
Would you mind adding a test for the error path too?

Reply:
@cameron-robey ping ^

Reply:
The error path still need a unit test.

Reply:
Added test in https://github.com/nodejs/undici/pull/1606/commits/0b4c50587ee1d77582e4702b1fec2af1ef0e9d86